Save session button

ciyer opened the pull request:
Provide a button save sessions.

## Testing scenarios

- [ ] Anonymous session (save should inform that user needs to log in and re-do their changes)
- [ ] Access level < Developer (save should inform that the user needs to fork the project and push their changes to the fork)
- [ ] No changes (save should inform the user that there are no changes to save)
- [ ] Behind origin (save should inform the user that they need to pull)
- [ ] Ahead of origin (save should let the user save, without a commit message)
- [ ] Dirty (save should provide a dialog box to enter a commit message)

## Screenshots

**Anonymous**
<img width="1288" alt="image" src="https://user-images.githubusercontent.com/1196411/194010000-33b5c30a-2efb-4e6b-9229-eb5911c0142b.png">


**Access level < Developer**
<img width="1288" alt="image" src="https://user-images.githubusercontent.com/1196411/194009113-85ca1de2-5a5c-4cfa-a48d-e3cc08bfded3.png">


**No changes**
<img width="1288" alt="image" src="https://user-images.githubusercontent.com/1196411/193999294-c93afe79-be79-4471-a786-8278e9e4d0bb.png">

**Behind origin**
<img width="1288" alt="image" src="https://user-images.githubusercontent.com/1196411/194007956-3f3fb512-ae74-4c09-aaf0-d46f69fe70ac.png">

**Ahead of origin**
<img width="1288" alt="image" src="https://user-images.githubusercontent.com/1196411/194008540-bc68e843-e859-43f0-9da3-b350436e4ff4.png">


**Dirty**
<img width="1288" alt="image" src="https://user-images.githubusercontent.com/1196411/194008222-eb2b8cbc-afc5-4215-9da8-4f5eb574cb08.png">

Fix #1985 

/deploy renku-notebooks=master #persist

ciyer commented:
Add as a comment to avoid causing the branch to re-deploy:

Fix #1985 

## Testing scenarios

- [ ] Anonymous session (save should inform that user needs to log in and re-do their changes)
- [ ] Access level < Developer (save should inform that the user needs to fork the project and push their changes to the fork)
- [ ] No changes (save should inform the user that there are no changes to save)
- [ ] Normal (save should provide a dialog box to enter a commit message)

andre-code reviewed (commented):
The code looks great! 👏
I see you create the services/session folder for the api, would it make sense to use the feature folder to keep all the APIs in the same place?

I've added a small suggestion regarding the save message in the modal stop session.
It would also be great to add e2e tests for all the cases implemented here.

ciyer commented on client/src/notebooks/components/StopSession.tsx:
```diff
@@ -55,7 +55,7 @@ function StopSessionModal({ stopSession, stopping, closeModal, isOpen }: StopSes
       Are you sure you want to stop this session?
       Make sure to commit any changes you want to be saved for next time
-      by running <code>renku save</code> or using Git.
+      (for example, by using the <b>Save</b> button).
```
Good suggestion. I have added this.

ciyer commented:
Thanks for the review. You are ab

> The code looks great! 👏 I see you create the services/session folder for the api, would it make sense to use the feature folder to keep all the APIs in the same place?
> 
> I've added a small suggestion regarding the save message in the modal stop session. It would also be great to add e2e tests for all the cases implemented here.

Thanks for the suggestions. I have made the changes to the code you suggest.

You are right that there should be e2e tests for this! Given the situation, though, I suggest we merge this without the e2e tests. #1986 will also interact with the sidecar, and I have added the requirement to create e2e tests in the description of that issue.

andre-code reviewed (approved):
lgtm
